lib/ansible/cli/doc.py

```diff
@@ -1245,7 +1245,16 @@ def get_man_text(doc, collection_name='', plugin_type=''):
                                 limit - 6, initial_indent=opt_indent[:-2] + "* ", subsequent_indent=opt_indent))
                     description = item.get('description', 'The official documentation on the %s module.' % item['module'])
                     text.append(textwrap.fill(DocCLI.tty_ify(description), limit - 6, initial_indent=opt_indent + '   ', subsequent_indent=opt_indent + '   '))
-                    text.append(textwrap.fill(DocCLI.tty_ify(get_versioned_doclink('modules/%s_module.html' % item['module'])),
```

Comment:
Looking at the links currently generated in `devel` and `stable-2.14`, it seems there's an existing issue with the URLs being created or perhaps an issue with the docs site itself. Either way, it's something we should fix.

This PR does seem to improve the situation, but it's not a labeled as a bug fix, so I'm assuming it wouldn't be backported. Using the `copy` module as an example, here's what I'm seeing:

This PR:

https://docs.ansible.com/ansible-core/devel/collections/ansible/builtin/assemble_module.html
https://docs.ansible.com/ansible-core/devel/collections/ansible/builtin/fetch_module.html
https://docs.ansible.com/ansible-core/devel/collections/ansible/builtin/file_module.html
https://docs.ansible.com/ansible-core/devel/collections/ansible/builtin/template_module.html
https://docs.ansible.com/ansible-core/devel/collections/ansible/posix/synchronize_module.html (missing)
https://docs.ansible.com/ansible-core/devel/collections/ansible/windows/win_copy_module.html (missing)

This PR when using `2.14` as the version:

https://docs.ansible.com/ansible-core/2.14/collections/ansible/builtin/assemble_module.html
https://docs.ansible.com/ansible-core/2.14/collections/ansible/builtin/fetch_module.html
https://docs.ansible.com/ansible-core/2.14/collections/ansible/builtin/file_module.html
https://docs.ansible.com/ansible-core/2.14/collections/ansible/builtin/template_module.html
https://docs.ansible.com/ansible-core/2.14/collections/ansible/posix/synchronize_module.html (missing)
https://docs.ansible.com/ansible-core/2.14/collections/ansible/windows/win_copy_module.html (missing)

From `devel` (all missing):

https://docs.ansible.com/ansible-core/devel/modules/ansible.builtin.assemble_module.html
https://docs.ansible.com/ansible-core/devel/modules/ansible.builtin.fetch_module.html
https://docs.ansible.com/ansible-core/devel/modules/ansible.builtin.file_module.html
https://docs.ansible.com/ansible-core/devel/modules/ansible.builtin.template_module.html
https://docs.ansible.com/ansible-core/devel/modules/ansible.posix.synchronize_module.html
https://docs.ansible.com/ansible-core/devel/modules/ansible.windows.win_copy_module.html

From `stable-2.14` (all missing):

https://docs.ansible.com/ansible-core/2.14/modules/ansible.builtin.assemble_module.html
https://docs.ansible.com/ansible-core/2.14/modules/ansible.builtin.fetch_module.html
https://docs.ansible.com/ansible-core/2.14/modules/ansible.builtin.file_module.html
https://docs.ansible.com/ansible-core/2.14/modules/ansible.builtin.template_module.html
https://docs.ansible.com/ansible-core/2.14/modules/ansible.posix.synchronize_module.html
https://docs.ansible.com/ansible-core/2.14/modules/ansible.windows.win_copy_module.html

There's also the question of why we're using `collections` for linking to built-in modules in the docs.

@felixfontein @samccann Do you have any insights on this?

Reply:
> Looking at the links currently generated in `devel` and `stable-2.14`, it seems there's an existing issue with the URLs being created or perhaps an issue with the docs site itself. Either way, it's something we should fix.

Yes, they were simply wrong so far.

> This PR does seem to improve the situation, but it's not a labeled as a bug fix, so I'm assuming it wouldn't be backported. Using the `copy` module as an example, here's what I'm seeing:

TBH I forgot I adjusted this line, I mainly fixed it because I copied this block for `plugin` and noticed it's also broken there.

IMO links should only be shown for `ansible.builtin` modules/plugins, since for modules/plugins outside `ansible.builtin` ansible-doc neither knows whether they are part of Ansible at all, and ansible-doc doesn't know the correct version of Ansible either.

> There's also the question of why we're using `collections` for linking to built-in modules in the docs.

This has been the case since the 2.10 docsite; all built-in modules and plugins are part of the synthetic collection `ansible.builtin`, so they are shown as members of that collection. For the ansible-core docsite, `ansible.builtin` is the only included collection (https://docs.ansible.com/ansible-core/devel/collections/index.html).